firebase_admin/_user_mgt.py

```diff
@@ -492,6 +599,58 @@ def get_user(self, **kwargs):
                 http_response=http_resp)
         return body['users'][0]
 
+    def get_users(self, identifiers):
+        """Looks up multiple users by their identifiers (uid, email, etc.)
+
+        Args:
+            identifiers: UserIdentifier[]: The identifiers indicating the user
+                to be looked up. Must have <= 100 entries.
+
+        Returns:
+            list[dict[string, string]]: List of dicts representing the JSON
+            `UserInfo` responses from the server.
+
+        Raises:
+            ValueError: If any of the identifiers are invalid or if more than
+                100 identifiers are specified.
+            UnexpectedResponseError: If the backend server responds with an
+                unexpected message.
+        """
+        if not identifiers:
+            return []
+        if len(identifiers) > 100:
+            raise ValueError('`identifiers` parameter must have <= 100 entries.')
+
+        payload = {}
+        for identifier in identifiers:
+            if isinstance(identifier, _user_identifier.UidIdentifier):
+                _auth_utils.validate_uid(identifier.uid, required=True)
```

Comment:
What if we push this validation and the subsequent marshaling to the corresponding classes?

Each identifier class can expose something like:

```
def marshal(self, payload):
   payload['localId'] = payload.get('localId', []).append(self.uid)
```

Reply:
I did this (just now) but didn't really like the results. It makes this function shorter, but it alters the `UserIdentifier` types such that they need to know how the json payload is constructed. (For instance, should it be `ProviderIdentifier`s job to know that `provider_uid` maps to `rawId` on the wire?) It also spreads building the payload across 2 files and 5 classes, instead of having it in just one spot.

One possible improvement we could make here is to eliminate the validation (or at least, refactor it somewhat). As per one of your other comments, I've already moved some validation to the ctors, but that won't stop users from changing the values after construction, but before calling get_users(), so we should keep the validation here. But if we altered (eg) UidIdentifier.uid to be UidIdentifier._uid and added a getter (essentially making these objects immutable) I think that could allow us to skip validation here entirely, which would allow this block to focus solely on creating the payload. I've done that. PTAL.

Reply:
I like how validation turned out (and the immutability of the properties). I don't feel strongly about the marshalling part. When I first suggested it, I felt that having each class being in charge of marshalling was a good thing. If we add more implementations of the UserIdentifier interface in the future, that's when this will matter more. I'm ok with leaving this as is for now.